### PR TITLE
Refactor `ContractCallServiceTest`

### PR DESCRIPTION
**Pull request by bilyana-gospodinova**
**Description**:
This PR refactors the rest of the tests in `ContractCallServiceTest` to use web3j. Some cleanup of unused methods is done in `ContractCallTestSetup`.

Fixes https://github.com/hashgraph/hedera-mirror-node/issues/8452

### CODE REVIEW

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/AbstractContractCallServiceTest.java [hunk 1]
```diff
@@ -69,24 +68,12 @@ public static Key getKeyWithContractId(final Contract contract) {
...
-                .withFailMessage(ESTIMATE_GAS_ERROR_MESSAGE, estimateGasUsedResult, actualGasUsed)
-                .isTrue();
-    }
-
     @BeforeEach
     void setup() {
         domainBuilder.recordFile().persist();
         testWeb3jService.setValue(0L);
         testWeb3jService.setSender(Address.fromHexString(""));
+        testWeb3jService.setBlockType(BlockType.LATEST);
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/AbstractContractCallServiceTest.java:76]: We should move these to `TestWeb3jService.reset()` and call it here.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java [hunk 2]
```diff
@@ -321,12 +299,6 @@ public class ContractCallTestSetup extends Web3IntegrationTest {
     @Value("classpath:contracts/NestedCallsTestContract/NestedCallsTestContract.json")
     protected Path NESTED_CALLS_ABI_PATH;
 
-    @Value("classpath:contracts/TestContractAddress/TestAddressThis.bin")
-    protected Path ADDRESS_THIS_CONTRACT_BYTES_PATH;
-
-    @Value("classpath:contracts/TestContractAddress/TestAddressThisInit.bin")
-    protected Path ADDRESS_THIS_CONTRACT_INIT_BYTES_PATH;
-
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java:329]: Can we delete any of the bin/json/sol files that you're removing from this file?

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java [hunk 3]
```diff
@@ -82,540 +99,472 @@ class ContractCallServiceTest extends ContractCallTestSetup {
...
-    }
+    @Autowired
+    private ContractExecutionService contractExecutionService;
 
+    @Override
     @BeforeEach
     void setup() {
         // reset gas metrics
+        super.setup();
         meterRegistry.clear();
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java:110]: Can remove this meterRegistry.clear() as it's done in a parent class already

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java [hunk 4]
```diff
@@ -82,540 +99,472 @@ class ContractCallServiceTest extends ContractCallTestSetup {
...
-                .flatMap(callType -> gasLimits.stream().map(gasLimit -> Arguments.of(callType, gasLimit)));
-    }
+    @Autowired
+    private ContractExecutionService contractExecutionService;
 
+    @Override
     @BeforeEach
     void setup() {
         // reset gas metrics
+        super.setup();
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java:109]: We should rename either child or parent setup() so we don't have to remember to manually invoke. Probably should make parent final, too.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java [hunk 5]
```diff
@@ -82,540 +99,472 @@ class ContractCallServiceTest extends ContractCallTestSetup {
...
-    }
-
-    private static Stream<Arguments> ercPrecompileCallTypeArgumentsProvider() {
-        List<Long> gasLimits = List.of(15_000_000L, 30_000L);
-
-        return Arrays.stream(CallType.values())
-                .filter(callType -> !callType.equals(ERROR))
-                .flatMap(callType -> gasLimits.stream().map(gasLimit -> Arguments.of(callType, gasLimit)));
-    }
+    @Autowired
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java:102]: Prefer standard `@Resource`

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java [hunk 6]
```diff
@@ -82,540 +99,472 @@ class ContractCallServiceTest extends ContractCallTestSetup {
...
-        assertThat(isWithinExpectedGasRange(
-                        longValueOf.applyAsLong(contractCallService.processCall(serviceParameters)), expectedGasUsed))
-                .isTrue();
-        assertGasLimit(serviceParameters);
+        verifyEthCallAndEstimateGas(functionCall, contract);
     }
 
     @Test
     void transferFunds() {
+        final var senderEntity = accountPersist();
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java:255]: Adding Given/When/Then to tests might aid in readability.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/AbstractContractCallServiceTest.java (hunk 1 above)
Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/AbstractContractCallServiceTest.java]: Moved

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java (hunk 2 above)
Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java:329]: Good point. I deleted the redundant files.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java (hunk 3 above)
Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java]: Comment addressed. Thanks.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java (hunk 4 above)
Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java]: After addressing the other comments, it seems that we don't need this `setup()` method in this class at all. I made the parent method final.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java (hunk 6 above)
Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceTest.java:277]: I added some comments. Hopefully the readability is better now.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/AbstractContractCallServiceTest.java [hunk 7]
```diff
@@ -69,24 +68,9 @@ public static Key getKeyWithContractId(final Contract contract) {
...
-                .withFailMessage(ESTIMATE_GAS_ERROR_MESSAGE, estimateGasUsedResult, actualGasUsed)
-                .isTrue();
-    }
-
     @BeforeEach
-    void setup() {
+    final void setup() {
         domainBuilder.recordFile().persist();
-        testWeb3jService.setValue(0L);
-        testWeb3jService.setSender(Address.fromHexString(""));
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/AbstractContractCallServiceTest.java:89]: Probably reset should occur in beforeeach, but can do in follow ups

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by kselveliev — APPROVED** [review]

**Review by IvanKavaldzhiev — APPROVED** [review]
LGTM

**Pull request merged**